### PR TITLE
Enable WebGPU testing on Linux

### PR DESCRIPTION
**Pull request by mattsoulanille**
Enable testing the WebGPU backend on Linux using Chrome Dev (105) instead of Chrome Canary (even though `base: 'ChromeCanary'` is listed, it launches Chrome Dev on Linux). Fix incorrect passing of flags to Chrome. Flags enabled in `chrome://flags` need to appear between `--flag-switches-begin` and `--flag-switches-end`.

Note that not all tests are passing on Linux. This could be due to Chrome Dev being behind Chrome Canary, or due to a lack of testing on Linux before this PR.

To see the logs from the Cloud Build CI, please join either our [discussion](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs) or [announcement](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs-announce) mailing list.

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/6591)

### CODE REVIEW

**Review by Linchenn — APPROVED** [review]

File: tools/karma_template.conf.js [hunk 1]
```diff
@@ -77,8 +77,11 @@ const CUSTOM_LAUNCHERS = {
   chrome_webgpu: {
     base: 'ChromeCanary',
     flags: [
+      '--disable-dawn-features=disallow_unsafe_apis',
+      '--flag-switches-begin',
       '--enable-unsafe-webgpu',
-      '--disable-dawn-features=disallow_unsafe_apis'
+      '--enable-features=Vulkan',
```
Comment (qjia7) [inline tools/karma_template.conf.js:83]: WebGPU on Linux requires command-line flag `--enable-features=Vulkan,UseSkiaRenderer`. So `UseSkiaRenderer` is missed? And this flag is only needed for Linux. We shouldn't add it on Windows. Can we distinguish them based on platforms?

Comment (mattsoulanille) [reply tools/karma_template.conf.js]: Thanks for the review! I added a [select](https://bazel.build/concepts/platforms-intro#select) to make sure the flag is added only on Linux. I also added `UseSkiaRenderer`.

File: tfjs-backend-webgpu/BUILD.bazel [hunk 2]
```diff
@@ -107,6 +107,9 @@ tfjs_web_test(
         '{"WEBGPU_CPU_FORWARD": false}',
     ],
     browsers = [],
-    local_browser = "chrome_webgpu",
+    local_browser = select({
+        "@bazel_tools//src/conditions:windows": "chrome_webgpu_windows",
```
Comment (qjia7) [inline tfjs-backend-webgpu/BUILD.bazel:111]: nit: Do we need to add the macOS?

**Review by qjia7 — APPROVED** [review]
LGTM with a nit. Thanks.

**Comment by mattsoulanille** [comment]
While testing this PR locally, I found out that "[webgpu] Replace module-scope 'let' with 'const' (#6602)" has caused most of the tests to fail on Linux (Chrome 105.0.5148.2). When it's reverted, only 8 tests fail. This could easily be related to my environment, but I thought I should mention it anyway.

File: tfjs-backend-webgpu/BUILD.bazel (hunk 2 above)
Comment (mattsoulanille) [reply tfjs-backend-webgpu/BUILD.bazel]: Good point. I renamed to Windows version to just `chrome_webgpu` and set it as the default condition. macOS will use it.

**Review by qjia7 — APPROVED** [review]
LGTM, thanks.

> While testing this PR locally, I found out that "[webgpu] Replace module-scope 'let' with 'const' (#6602)" has caused most of the tests to fail on Linux (Chrome 105.0.5148.2). The failure mode is that all the values returned are zero. When it's reverted, only 8 tests fail. This could easily be related to my environment, but I thought I should mention it anyway.

Thanks Matt. The `wgsl` spec change of using module-scope `const` instead of `let` is a very new change. So I suppose you need to update your chrome. My chrome canary is 105.0.5164.0. And can you help to file a bug for the 8 tests failures? We will take a look, thanks

**Comment by gyagp** [comment]
On Linux, the most frequently updated channel is Dev, and it doesn't include the latest WebGPU changes. Maybe you need to wait for a few days for the recent breaking changes to be rolled into Dev. 
We just enabled the tests on Linux today, and @xhcao will take care of all the failures there.

**Pull request merged**